java-security/src/main/java/com/sap/cloud/security/token/validation/validators/XsuaaJwtSignatureValidator.java

```diff
@@ -54,12 +56,10 @@ private PublicKey fetchPublicKey(Token token, JwtSignatureAlgorithm algorithm) t
             throw new IllegalArgumentException("Token does not contain the mandatory " + KID_PARAMETER_NAME + " header.");
         }
 
-        String jwksUri = configuration.isLegacyMode() ? configuration.getUrl() + "/token_keys" : token.getHeaderParameterAsString(JKU_PARAMETER_NAME);
-        if (jwksUri == null) {
-            throw new IllegalArgumentException("Token does not contain the mandatory " + JKU_PARAMETER_NAME + " header.");
-        }
-
+        String jwksUri = configuration.isLegacyMode() ? configuration.getUrl() + "/token_keys" : configuration.getProperty(UAA_DOMAIN) + "/token_keys";
+        URI uri = URI.create(jwksUri);
+        uri =  uri.isAbsolute() ? uri : URI.create("https://" + jwksUri);
```

Comment:
We also need to put the zid from the token as query parameter into the URL, e.g. .../token_keys?zid=<tokenZid>.
If it is not in the token or blank, this query parameter can be omitted.